src/services/ui/src/pages/detail/package-activity/hook.ts

```diff
@@ -0,0 +1,63 @@
+import { getAttachmentUrl } from "@/api";
+import JSZip from "jszip";
+import { saveAs } from "file-saver";
+import { useState } from "react";
+import { opensearch } from "shared-types";
+
+type Attachments = NonNullable<opensearch.changelog.Document["attachments"]>;
+
+export const useAttachmentService = (
+  props: Pick<opensearch.changelog.Document, "packageId">
```

Comment:
There is some redundancy between the prop and the function signatures within the hook.

The main reason this requires the top level changelog document is for access to the packageId.  In addition, it provided consistent props scaffolding.